lib/active_merchant/billing/gateways/blue_snap.rb

```diff
@@ -204,6 +205,41 @@ def add_address(doc, options)
         doc.zip(address[:zip]) if address[:zip]
       end
 
+      def add_level_3_data(doc, options)
+        return unless options[:customer_reference_number]
```

Comment:
Added return condition here to prevent empty `level-3-data` container tag.

Reply:
Going to ask the obvious, can `customer_reference_number` be excluded but 1 or more of the other fields provided, where we should send L3 data elements?

Reply:
Good q- `customer_reference_number` and `sales_tax_amount` are the only fields required by all 3 processing types: Level 3, Level 2, Level 2 with TAA. There are no processing types that are valid without both of those.

Reply:
(Level 3 requires several more in addition)

Reply:
👍